File: rockpaperscissors.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

int main() {
    srand(time(0));
    std::string choices[] = {"rock", "paper", "scissors"};
    std::string computer = choices[rand() % 3];
    std::string player;

    std::cout << "Enter rock, paper, or scissors: ";
    std::cin >> player;

    std::cout << "Computer chose " << computer << std::endl;

    if (player == computer) {
        std::cout << "Tie!" << std::endl;
    } else if ((player == "rock" && computer == "scissors") ||
               (player == "paper" && computer == "rock") ||
               (player == "scissors" && computer == "paper")) {
        std::cout << "You win!" << std::endl;
    } else {
        std::cout << "You lose!" << std::endl;
    }

    return 0;
}

File: rockpaperscissors.java
import java.util.Random;
import java.util.Scanner;

public class rockpaperscissors {
    public static void main(String[] args) {
        String[] choices = {"rock", "paper", "scissors"};
        Random rand = new Random();
        String computer = choices[rand.nextInt(3)];
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter rock, paper, or scissors: ");
        String player = scanner.nextLine().toLowerCase();

        System.out.println("Computer chose " + computer);

        if (player.equals(computer)) {
            System.out.println("Tie!");
        } else if ((player.equals("rock") && computer.equals("scissors")) ||
                   (player.equals("paper") && computer.equals("rock")) ||
                   (player.equals("scissors") && computer.equals("paper"))) {
            System.out.println("You win!");
        } else {
            System.out.println("You lose!");
        }
        scanner.close();
    }
}